homeassistant/components/stiebel_eltron/climate.py

```diff
@@ -0,0 +1,153 @@
+"""Support for stiebel_eltron climat platform."""
+import logging
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_HUMIDITY, STATE_AUTO, STATE_IDLE, STATE_MANUAL,
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.components.stiebel_eltron import DOMAIN as STE_DOMAIN
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+DEPENDENCIES = ['stiebel_eltron']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
+
+STATE_DAYMODE = 'day_mode'
+STATE_SETBACK = 'setback_mode'
+STATE_DHW = 'dhw'
+STATE_EMERGENCY = 'emergency'
+
+"""Mapping STIEBEL ELTRON states to homeassistant states."""
+STE_TO_HA_STATE = {'AUTOMATIC': STATE_AUTO,
+                   'MANUAL MODE': STATE_MANUAL,
+                   'STANDBY': STATE_IDLE,
+                   'DAY MODE': STATE_DAYMODE,
+                   'SETBACK MODE': STATE_SETBACK,
+                   'DHW': STATE_DHW,
+                   'EMERGENCY OPERATION': STATE_EMERGENCY}
+
+"""Mapping homeassistant states to STIEBEL ELTRON states."""
+HA_TO_STE_STATE = {value: key for key, value in STE_TO_HA_STATE.items()}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the StiebelEltron platform."""
+    name = hass.data[STE_DOMAIN]['name']
+    ste_data = hass.data[STE_DOMAIN]['ste_data']
+
+    add_devices([StiebelEltron(name, ste_data)], True)
+    return True
+
+
+class StiebelEltron(ClimateDevice):
+    """Representation of a STIEBEL ELTRON heat pump."""
+
+    def __init__(self, name, ste_data):
+        """Initialize the unit."""
+        self._name = name
+
+        self._target_temperature = None
+        self._current_temperature = None
+        self._current_humidity = None
+        self._operation_modes = [STATE_AUTO, STATE_IDLE, STATE_DHW]
+        self._current_operation = None
+        self._filter_alarm = None
+        self._ste_data = ste_data
+
+        _LOGGER.debug("Initialized climate platform")
```

Comment:
We try to avoid side effects in init method. Please remove the logging. This message is already logged by the core.

Reply:
Done